### DISCUSSION TITLE
Ecolink TBZ500 Thermostat (works with Home Assistant ootb).

### DISCUSSION THREAD

**Discussion by GrizzlyAK** [zwave-js/zwave-js#3714] [General]
Although this device doesn't appear in the ZWave JS Config DB Browser, I wanted to confirm that I have successfully added it to Home Assistant, with S2-Unauth, and it was found by HA, climate entities created, and worked immediately using a Climate Lovelace Card. It is a very nice, full-featured thermostat and completely cloud-free.

The only things I've noticed are the following:

(1) The Manufacturer was specified as "0x5442" and Product was specified as "Unknown product 0x5439" in the zwavejs2mqtt CP.

(2) In the back of the manual [here](https://discoverecolink.com/wp-content/uploads/PRODUCT-SHEET-TBZ500-THERMOSTAT-RDN1150921.pdf), there are a TON of configuration parameters that do not appear in my zwavejs2mqtt CP. In fact, there is no Configuration block in the CP for the Node. I suspect it has yet to be defined here.

Is there anything I can do to assist with either of these.

**Comment by AlCalzone** [comment]
This is completely expected. The device probably uses Configuration CC V1 or 2, which don't allow discovering the parameters automatically. Therefore they have to be defined in a config file.

> Is there anything I can do to assist with either of these.

Yes, write such a config file and provide it in a PR, see https://zwave-js.github.io/node-zwave-js/#/config-files/contributing-files

**Comment by GrizzlyAK** [comment]
It's a new device, only approved in May this year, I think. According to the manual, it uses Config CC V2.

I had already looked at the docs you linked to, and although I regularly modified the local XML files for the old OZW in HA for devices that didn't have them, I don't know if I have the requisite knowledge to create such a file. I'm sure I could get some of it right, but there are many I don't understand. I checked the links in the docs to see if a file exists on other systems, and none had it. The alliance web site shows only 2 parameters, whereas the manual list 71.

I'd be happy to give it a shot and let someone more knowledgeable finish it if that would help get one created.

**Reply by AlCalzone** [reply]
Go for it! The docs are pretty extensive and should explain all the parts of the config files.
If there are any questions, feel free to ask.

**Reply by GrizzlyAK** [reply]
@AlCalzone if I have readOnly parameters, and the values are defined, do I add the options: array and list them as you would with writeOnly? 

Also, what's the general guidance on the use of `unsigned:`. Most parameters are positive only, since it defaults to `false`, is it expected pretty much everywhere? 

Finally, are numbers for values decimal only, or is hex allowed?

**Reply by GrizzlyAK** [reply]
PR  submitted.

**Reply by AlCalzone** [reply]
> if I have readOnly parameters, and the values are defined, do I add the options: array and list them

Yes, this allows UIs to give the numbers a meaningful label.

> Also, what's the general guidance on the use of unsigned:.

If the values are positive and within the range of signed values, it doesn't matter if you specify that. If it matters, the checks will fail and tell you to use `unsigned`.

> are numbers for values decimal only, or is hex allowed?

Hex would be allowed by the parser, but the JSON schema we have to give context help in VSCode doesn't support it, so better stick to decimal values.

**Reply by GrizzlyAK** [reply]
Copy that on everything.  I will go back and add the options for the RO values for those two parameters. Thanks for the info and clarification. And the checks DID catch one of those unsigned errors where I had set the valueSize too small.

**Reply by derekwolfson** [reply]
Did you all end up finishing the PR?

**Reply by GrizzlyAK** [reply]
Yes. https://github.com/zwave-js/zwave-js/pull/3731

**Reply by derekwolfson** [reply]
Awesome!  Just about to build a system with this included!!

**Comment by GrizzlyAK** [comment]
Mine is still working great. The only thing I don’t like about them is that the temp settings are in 1 deg increments (mine was given to me by Ecolink to evaluate 4-5 years ago when it first came out, so that may have changed by now). My four other virtual thermostats use Zooz ZEN16 multirelays and various temperature sensors around the house and can be as accurate as 0.1 degree. It works great. Note that I did leave the original mercury switch thermostats connected in parallel set to 50 to serve as safety mechanisms in case of HA failure during the winter (I’m in AK). This TS in HA looks just like my manual ones and they all work the same, including temp history charts. The only difference is the increment. Good luck! Sent from my iPhoneOn Nov 29, 2025, at 9:09 AM, derekwolfson ***@***.***> wrote:
Awesome!  Just about to build a system with this included!!
